### ISSUE TITLE
Add Forward Delete key

### ISSUE THREAD

**Issue by jonbarh** [florisboard/florisboard#2042]
It would be nice to have the option of asign that function to and existing key, like utility key, or display it besides <--Del. It is useful to correct long texts or to avoid to move the cursor to the right spot.
Thank you very much.

**Comment by jonbarh** [comment]
It would be nice to have the option of asign that function to and existing key, like utility key, or display it besides <--Del. It is useful to correct long texts or to avoid to move the cursor to the right spot.
Thank you very much.

**Comment by tsiflimagas** [comment]
It sounds useful, but I'm not sure whether it is worthy to be added, since you can just select text holding Shift and either glide on the spacebar or use the arrows on the smartbar, then hit delete. Would substitute this feature sufficiently. Note that it's just my opinion, perhaps adding this feature doesn't bring any redundancy.

**Comment by patrickgold** [comment]
In general the forward delete button (that's what it is called I think) is a really useful key, especially on PC. In the internal codebase it is somewhat implemented, however there's no assigned UI key(button to it. I plan to add that key in the special keys layout, maybe in an advanced PC layout, and by allowing users to place this key anywhere once I implement in-app layout customization.

So I guess we can keep this issue open, although I could have sworn that there's already a proposal for this, but I cannot find that issue at all atm. (Kostas if you should find it you can mark this as a duplicate, else we will use this issue for the "Forward Delete" key).

**Comment by jonbarh** [comment]
I tried several keyboards and found this funtionality only implemented by defect in a few, some Samsung tablet, located as it is in PC (at least in spanish layout, called "Supr") in the upper right corner, fixed. For years I used one called MultilingOKboard because it offers the option to define a custom layout by editing the JSON files, so I set a Forward Del key next the <--Del key.
OKBoard development seems mostly halted and it has some bugs related to layout customization so, for a couple of months now, I've been using Floris beta as my principal keyboard. So yes indeed I do what Kostas suggest: moving cursor using space bar swipe till the desired position and then delete. The importance of the Forward Del key for me is that, for chat, the suggestions works very well and, abow all, are fast, but for longer texts, posts, etc, that needed more complex corrections and often you are checking it from the beggining several times, this function is key because you are changing and deleting a lot of caracters, words and phrases, after de cursor and moving it each time is exhausting and multiplies the needed keystrokes. This text of mine here is and example of that since english is not my language. Also, with Forward Del key you can blindly delete the text til the end.

Thank you very much for you work, Floris is, even in its current development state, an excellent and cleverly designed keyboard.

**Comment by thornySoap** [comment]
I'm not sure whether I should rather open a new proposal or if a similar already exists. But I think I once tested a keyboard that deleted forward when holding shift und pressing the usual delete button. I personally couldn't fit an extra key into the layout but it would be quite handy if the shift key could sort of invert the delete direction.

**Comment by RokeJulianLockhart** [comment]
I need this for [`mail.proton.me/u/0/inbox`](https://mail.proton.me/u/0/inbox/xPfQ4whNwCCwDe9tmbImbgxOXrDpW-Db-s1ZskUqNKYApdo9piubqsTx_-32pWs69UieSlO8G_KtGqDbJ0V7sw==), because my signature has `h4`s whose automatically prefixed `\n`s must be deleted. Using backspace on the start of the heading line instead converts the `h4` into a `p` tag, which isn't semantic.

**Comment by mysteryoneal** [comment]
Yes, please, I miss this feature from the days when I was using MultilingO. While, yes, technically you can use highlighting to batch delete and other things like that. None of those options are anywhere near as efficient or as accurate as using the forward delete key. It comes in especially useful when trying to delete everything forward of a point for large batches when combined with whole word delete and long press functionality.

It has been a couple of years since it has been stated that this is partially implemented already. Is there any clue when customizable layouts will become functional?

**Comment by Perdellian** [comment]
> I'm not sure whether I should rather open a new proposal or if a similar already exists. But I think I once tested a keyboard that deleted forward when holding shift und pressing the usual delete button. 

#450 proposes shift+delete as well, albeit referring to the key as capslock.

While a discrete button could also be nice, I think shift+del better of the two since, as mentioned in [this comment](https://github.com/florisboard/florisboard/issues/464#issuecomment-808936781) on #464, the concept could also be applied to reversing the direction of swipe deletion.

**Comment by patrickgold** [comment]
There is already another older issue for this, #450. Closing this issue in favor of the older one. See details there regarding implementation idea.

**Comment by mysteryoneal** [comment]
This was not a duplicate. It was a similar but separate feature request.
The most recent comment WAS a duplicate of the one that you were claiming
this one was a duplicate of. But this thread is about adding a separate
forward delete key, not some key combination to add functionality to the
existing delete key.

**Comment by patrickgold** [comment]
Hmm, true, for now I will focus on the combined delete key issue though

**Comment by patrickgold** [comment]
PR #3010 adds an explicit forward delete quick action, which resolves this issue. Adding the forward delete key within the layout itself is theoretically possible but a point for custom layouts, for which separate issues exist (when custom layout support is added all keys are supported anyways).
